server/neptune/workflows/internal/terraform/workflow.go

```diff
@@ -127,7 +127,7 @@ func (r *Runner) Plan(ctx workflow.Context, root *terraform.LocalRoot, serverURL
 			// not returning UpdateJobError here since we want to surface the job failure itself
 			logger.Error(ctx, "unable to update job with failed status, job failed with error. ", "err", err)
 		}
-		return response, errors.Wrap(err, "running job")
+		return response, newTerraformClientError(err, "running job")
```

Comment:
Isn't it possible that this is not a terraform client error and is another error returned from the runner/activity?

I'm not sure we should approach it like this, instead consider looking for the specific TerraformClientError thrown from the activity and then if that's present, wrapping this our external facing ClientError.